src/main/java/pl/damianszczepanik/jenkins/buildhistorymanager/BuildHistoryManager.java

```diff
@@ -48,25 +48,29 @@ public void perform(Job<?, ?> job) throws IOException, InterruptedException {
 
         Run<?, ?> run = job.getLastCompletedBuild();
         // for each completed build...
-        do {
+        while (run != null) {
             LOG.info("Processing build #" + run.getNumber());
-            for (int i = 0; i < rules.size(); i++) {
-                Rule rule = rules.get(i);
-                LOG.info("Processing rule no " + (i + 1));
-                if (rule.validateConditions(run)) {
-                    LOG.info("Processing actions for rule no " + (i + 1));
-                    rule.performActions(run);
+            if (run.isKeepLog()) {
```

Comment:
yes but still not here

Reply:
what do you mean?

Reply:
I read again as I though it keeps only log not the whole build.

I can imagine that someone uses this plugin to rename the build or add description so then this flag has no sense but let's keep it as it is.

Reply:
As far as I remember it keeps everything from the build.
Btw thanks for your help here!

Reply:
So what do you exactly want here?

Reply:
> from my experience checking `isKeepLog()` is the only way to check with the given api if the user marked a build from the ui as keep forever.
> 
> ![image](https://user-images.githubusercontent.com/14205420/106365910-e074c700-6338-11eb-8e8d-83058a2042ee.png)
> 
> In my opinion no one who does this is willing to lose artifacts.
> What do you propose instead? Introducing a new condition which matches build that are marked as keep forever?

In my case I mark my versionbuilds as "keep forever" and the first time I used the plugin all my important version builds and the binaries and installers were deleted.

If this happens to a new user which is not using a test system this is fatal.